src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/mc_sseg_stime_NOLOOP_demo.py

```diff
@@ -90,33 +102,73 @@ def main():
     cs = 1.399999976158142  # channel trapezoidal sideslope
     s0 = 0.0017999999690800905  # downstream segment bed slope
     qlat = 40.0  # Lateral inflow in this time step
-    qup = (
-        0.04598825052380562  # Flow from the upstream neighbor in the previous timestep
-    )
-    quc = 0.04598825052380562  # Flow from the upstream neighbor in the current timestep
-    qdp = 0.21487340331077576  # Flow at the current segment in the previous timestep
-    depthp = (
-        0.010033470578491688  # Depth at the current segment in the previous timestep')
-    )
-    velp = 0.070480190217494964  # Velocity in the current segment in the previous timestep NOT USED AS AN INPUT!!!
-    """
-    k,   i,   q,    vel,    depth
-    0 0 0.0891760066151619 0.06941037625074387 0.010050795041024685
-    0 1 0.06863606721162796 0.07104580104351044 0.009998836554586887
-    0 2 0.04598825052380562 0.06900732219219208 0.009981763549149036
-    0 3 0.21487340331077576 0.07048019021749496 0.010033470578491688
-    1 0 0.3173820674419403 0.15248453617095947 0.03275299444794655
-    1 1 0.2014089822769165 0.09322302788496017 0.015030190348625183
-    1 2 0.13257014751434326 0.061184611171483994 0.008333344012498856
-    1 3 0.7570106983184814 0.12373604625463486 0.02334451675415039
-    """
-
-    qdc_expected = 0.7570106983184814
-    velc_expected = 0.12373604625463486
-    depthc_expected = 0.02334451675415039
+
+    if precision is "single":
+
+        """
+        single precision results from standard procedure
+        ORIG..ORIG
+        k,   i,   q,    vel,    depth
+        0 0 0.0891760066151619 0.06941037625074387 0.010050795041024685
+        0 1 0.06863606721162796 0.07104580104351044 0.009998836554586887
+        0 2 0.04598825052380562 0.06900732219219208 0.009981763549149036
+        0 3 0.21487340331077576 0.07048019021749496 0.010033470578491688
+        1 0 0.3173820674419403 0.15248453617095947 0.03275299444794655
+        1 1 0.2014089822769165 0.09322302788496017 0.015030190348625183
+        1 2 0.13257014751434326 0.061184611171483994 0.008333344012498856
+        1 3 0.7570106983184814 0.12373604625463486 0.02334451675415039
+        """
+        # Providing the truncated inputs produces the exact same answer...
+        qup = 0.04598825
```

Comment:
I think this should use the ctypes.c_float type. qup is still a double precision float here (even though it is truncated).
```
import ctypes
qup = ctypes.c_float(0.04598825052380562)
```

Reply:
Hmm, f2py doesn't seem to like ctypes objects.
```
ValueError: mc_sseg_stime.muskingcunge_module.muskingcungenwm() 1st argument (dt) can't be converted to float
```

Reply:
Based on your experiment here, I think we're moving forward by forcing the precision in the fortran to single precision. We can maybe take a second step to make sure that the communication between the modules is explicitly single or double precision.